src/nextcloud/api_wrappers/webdav.py

```diff
@@ -102,30 +103,42 @@ def download_file(self, uid, path):
         with open(filename, 'wb') as f:
             f.write(res.data)
 
-    def upload_file(self, uid, local_filepath, remote_filepath):
+        # set timestamp of downloaded file
+        file_timestamp_str = (file_data.data[0].get('last_modified'))
+        try:
+            file_timestamp = datetime.strptime(file_timestamp_str, '%a, %d %b %Y %H:%M:%S GMT')
```

Comment:
Please extract this "timestamp str to timestamp" to a separate function, and use that one in tests.